pyrdp/parser/rdp/orders/frontend.py

```diff
@@ -0,0 +1,182 @@
+#
+# This file is part of the PyRDP project.
+# Copyright (C) 2019 GoSecure Inc.
+# Licensed under the GPLv3 or later.
+#
+"""
+Drawing Order Context.
+"""
+
+from pyrdp.parser.rdp.orders.alternate import CreateOffscreenBitmap, SwitchSurface, CreateNineGridBitmap, \
+    StreamBitmapFirst, StreamBitmapNext, GdiPlusFirst, GdiPlusNext, GdiPlusEnd, GdiPlusCacheFirst, \
+    GdiPlusCacheNext, GdiPlusCacheEnd, FrameMarker
+
+from .common import Bounds
+from pyrdp.enum import IntEnum
+
+
+class BrushStyle(IntEnum):
+    SOLID = 0x00
+    NULL = 0x01
+    HATCHED = 0x02
+    PATTERN = 0x03
+
+
+class HatchStyle(IntEnum):
+    HORIZONTAL = 0x00
+    VERTICAL = 0x01
+    FDIAGONAL = 0x02
+    BDIAGNOAL = 0x03
+    CROSS = 0x04
+    DIAGCROSS = 0x05
+
+
+class GdiFrontend:
+    """
+    Interface for objects that implement GDI.
+
+    This class provides abstract methods to be used by modules
+    interested in listening and acting upon context updates.
+    Its primary purpose is for the PyRDP player to render the
+    remote desktop.
+
+    NOTE: Unimplemented methods will act as No-Op.
```

Comment:
shouldnt they raise or at least log?

Reply:
Not for the base class. The current frontend implementation `GdiFrontend` warns on the first unsupported order it sees.